service/internal/s3shared/endpoint_error.go

```diff
@@ -20,27 +20,27 @@ type InvalidARNError struct {
 	origErr  error
 }
 
-// Error returns the InvalidARNError
+// Code returns the invalid ARN error code
+func (e InvalidARNError) Code() string {
+	return invalidARNErrorErrCode
+}
+
```

Comment:
We don't want to expose this concept of a Code on non service API errors. This should only be `Error` and `Unwrap` and optionally any sort of "getter" type accessors if the error is wrapping and wanting to provide additional context.